docs/reference/glossary.md

```diff
@@ -0,0 +1,200 @@
+
+```hcl title="communication/send-sms.twilio.suma"
+profile = "communication/send-sms@2.0"
+provider = "twilio"
+
+map SendMessage {
+  http POST "/2010-04-01/Accounts/{parameters.accountSid}/Messages.json" {
+    security "basic"
+
+    request "application/x-www-form-urlencoded" {
+      body {
+        To = input.to,
+        From = input.from,
+        Body = input.text
+      }
+    }
+
+    response 201 "application/json" {
+      map result {
+        messageId = body.sid
+      }
+    }
+  }
+}
+
+map RetrieveMessageStatus {
+  // body of the RetrieveMessageStatus logic
+}
+```
+
+## Provider {#provider}
+
+_Provider_ or _integration provider_ defines a set of base URLs (services), security schemes, and [integration parameters](#integration-parameter). This information is used for reusability between maps and as a security measure. Maps cannot send requests outside of services defined and cannot read values passed to the authentication scheme.
+
+:::info Provider vs. company
+
+From Superface perspective, single provider doesn't have to correspond to a single company. Large companies, like Google, have multiple unrelated APIs, each with its own provider, e.g. `google-apis-maps` or `google-apis-computer-vision`. On the other hand, your provider can specify services from unrelated companies if you need to use these services in a single map.
+
+:::
+
+Provider definition is a JSON file adhering to the [Comlink Provider schema](#comlink). The provider is _configured_ by the Superface CLI for particular profiles, which enables the respective maps to be used by [OneSDK](#one-sdk) in the runtime. Provider is identified by its name, for example: `twilio`.
+
+:::caution Provider name limitations
+
+Name of the provider cannot contain numbers to avoid typosquatting attacks. If you are publishing a provider into the registry, its name must start with `unverified-` prefix, e.g. `unverified-twilio`. See [Verification](../verification.mdx) for further details.
+
+:::
+
+Example of the `twilio` provider:
+
+```json title="twilio.provider.json"
+{
+  "name": "twilio",
+  "services": [
+    {
+      "id": "default",
+      "baseUrl": "https://api.twilio.com"
+    }
+  ],
+  "defaultService": "default",
+  "securitySchemes": [
+    {
+      "id": "basic",
+      "type": "http",
+      "scheme": "basic"
+    }
+  ],
+  "parameters": [
+    {
+      "name": "accountSid",
+      "description": "Account security identifier"
+    }
+  ]
+}
+```
+
+## Registry {#registry}
+
+_Registry_ is a web service for distribution and discovery of profiles, maps, and providers. The service accepts [Comlink](#comlink) files for publishing and generates Abstract Syntax Tree (AST) for runtime consumption by [OneSDK](#one-sdk).
+
+The public registry is available at [superface.ai](https://superface.ai/catalog). Integrations available in the public catalog are maintained and published from the [Station GitHub repository](https://github.com/superfaceai/station).
+
```

Comment:
I think _Registry_ section can be moved under _Integration parameter_.

It will make the transitions between _Provider → Parameter_ and _Registry → OneSDK_ more fluent I guess. I know this is not necessarily a continuous text, but reading it that way felt weird due to that order.